lib/usecases/get_current_survey_use_case.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:convert';
+import 'package:injectable/injectable.dart';
+import 'package:survey_flutter_ic/api/storage/storage.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
+
+import 'base/base_use_case.dart';
+
+@Injectable()
+class GetCurrentSurveyUseCase extends NoParamsUseCase<SurveyModel?> {
+  final Storage _storage;
```

Comment:
As far as I understand the clean architecture, the use case should only access the repository to request the action, not the service. So I suggest we create a method to get the `currentSurvey` in repository and then use it here

<img width="600" src="https://github.com/nimblehq/flutter-ic-david-bruce/assets/25218255/c99319b5-8bf3-41a2-b121-09305a58fbf2">

Reply:
@thiennguyen0196  🤔 Could you please explain more about this comment? I think don't get your idea clearly.

Reply:
@nmint8m Because our app are following Clean architecture, so every use case should only have access to the repository and domain model, and the services (Api service, DB service) will only be accessed by repositories as the above image shown.
So the level of access should be `use cases -> repositories -> db/api service`
😄